app/views/puppetclasses/_selectedClasses.html.erb

```diff
@@ -1,4 +1,4 @@
-<%= content_tag_for :li, klass, :selected, :class => cycle('even', 'odd') do %>
+<%= content_tag_for :li, klass, :selected, :class => [cycle('even', 'odd'), ('unavailable' unless klass.environments.include?(@environment))] do %>
```

Comment:
Found the issue. `@environment` is only available on the *first* load of the page. On subsequent loads, it doesn't exist as the host controller didn't set it.

I fixed it locally by adding two lines to `hostgroup_or_environment_selected`, so it's the following:

```ruby

  def hostgroup_or_environment_selected
    Taxonomy.as_taxonomy @organization, @location do
      if params['host']['environment_id'].present? || params['host']['hostgroup_id'].present?
        @environment = Environment.find(params['host']['environment_id'])
        @hostgroup = Hostgroup.find(params['host']['hostgroup_id'])
        render :partial => 'puppetclasses/class_selection', :locals => {:obj => (refresh_host)}
      else
        logger.info "environment_id or hostgroup_id is required to render puppetclasses"
      end
    end
  end
```

I think that's all that's required for it to work.